### ISSUE TITLE
vNext insider BC: "The property 'Value' cannot be found on this object. Verify that the property exists and can be set." during container creation

### ISSUE THREAD

**Issue by kine** [microsoft/navcontainerhelper#2874]
**Describe the issue**
When creating container for current insider vNext versionm the creation ends with this error>
```
The property 'Value' cannot be found on this object. Verify that the property exists and can be set.
at <ScriptBlock>, C:\Run\SetupConfiguration.ps1: line 106
at <ScriptBlock>, C:\Run\navstart.ps1: line 155
at <ScriptBlock>, C:\Run\start.ps1: line 384
at <ScriptBlock>, <No file>: line 1Error
```

**Scripts used to create container and cause the issue**
```
$containerName = 'test'
Windows credentials for the host computer.'
$auth = 'AAD'
$sasToken = 'xxxx'
$artifactUrl = Get-BcArtifactUrl -storageAccount 'bcinsider' -type 'Sandbox' -country 'us' -select 'Latest' -sasToken $sasToken
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -authenticationEMail 'xxx@xxxx.xxx' `
    -updateHosts

```

**Full output of scripts**
```
BcContainerHelper is version 4.0.8
BcContainerHelper is running as administrator
Hyper-V is Enabled
UsePsSession is True
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 20.10.22
Docker Server Version is 20.10.22
Removing Desktop shortcuts
Downloading application artifact /sandbox/22.0.52293.0/us
https://bcinsider.azureedge.net/sandbox/22.0.52293.0/us?xxxxxxx
Downloading C:\Users\xxxxx\AppData\Local\Temp\c2c252a9-0df4-4c7b-a609-8ff06e89086d.zip
Using WebClient
Unpacking application artifact to tmp folder using 7zip
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:10.0.17763.3770
Creating Container test
Style: sandbox
Multitenant: Yes
Version: 22.0.52293.0
Platform: 22.0.52192.0
Generic Tag: 1.0.2.13
Container OS Version: 10.0.17763.3770 (ltsc2019)
Host OS Version: 10.0.17763.3770 (ltsc2019)
Using process isolation
Using locale en-US
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\NavContainerHelper\Extensions\test\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container test from image mcr.microsoft.com/businesscentral:10.0.17763.3770
aaf99bb4f553f198f5c9dde4d2df86346ed8e778ca1f2e9218efaa5eb5624e7a
Waiting for container test to be ready
Using artifactUrl https://bcinsider.azureedge.net/sandbox/22.0.52293.0/us
Using installer from C:\Run\210-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\22.0.52293.0\platform\ServiceTier\Program Files
c:\dl\sandbox\22.0.52293.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\22.0.52293.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\22.0.52293.0\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\sandbox\22.0.52293.0\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\22.0.52293.0\platform
c:\dl\sandbox\22.0.52293.0\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\22.0.52293.0\us\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\22.0.52293.0\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\22.0.52293.0\us\Extensions
Copying Applications
C:\dl\sandbox\22.0.52293.0\platform\Applications
Copying Applications.US
C:\dl\sandbox\22.0.52293.0\us\Applications.US
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 431 seconds
Installation complete
Initializing...
Setting host.containerhelper.internal to 172.28.64.1 in container hosts file
Starting Container
Hostname is test
PublicDnsName is test
Using AccessControlService Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint ACB630C07A810C6CD5B377B631E0BA8D24BA9784
DNS identity test
Modifying Service Tier Config File with Instance Specific Settings
The property 'Value' cannot be found on this object. Verify that the property exists and can be set.
at <ScriptBlock>, C:\Run\SetupConfiguration.ps1: line 106
at <ScriptBlock>, C:\Run\navstart.ps1: line 155
at <ScriptBlock>, C:\Run\start.ps1: line 384
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcinsider.azureedge.net/sandbox/22.0.52293.0/us
Using installer from C:\Run\210-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\22.0.52293.0\platform\ServiceTier\Program Files
c:\dl\sandbox\22.0.52293.0\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\22.0.52293.0\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\22.0.52293.0\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\sandbox\22.0.52293.0\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\22.0.52293.0\platform
c:\dl\sandbox\22.0.52293.0\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\22.0.52293.0\us\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\22.0.52293.0\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\22.0.52293.0\us\Extensions
Copying Applications
C:\dl\sandbox\22.0.52293.0\platform\Applications
Copying Applications.US
C:\dl\sandbox\22.0.52293.0\us\Applications.US
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 431 seconds
Installation complete
Initializing...
Setting host.containerhelper.internal to 172.28.64.1 in container hosts file
Starting Container
Hostname is test
PublicDnsName is test
Using AccessControlService Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint ACB630C07A810C6CD5B377B631E0BA8D24BA9784
DNS identity test
Modifying Service Tier Config File with Instance Specific Settings
The property 'Value' cannot be found on this object. Verify that the property exists and can be set.
at <ScriptBlock>, C:\Run\SetupConfiguration.ps1: line 106
at <ScriptBlock>, C:\Run\navstart.ps1: line 155
at <ScriptBlock>, C:\Run\start.ps1: line 384
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: ee906cc2-33e4-4cbf-95bb-8b0e5649dda1
Initialization of container test failed
At C:\Users\adm_ksacek\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.8\ContainerHandling\Wait-NavContainerReady.ps1:42 char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization of container test failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container test failed
 ```

**Additional context**
The problem is that the script is trying to set config value ClientServicesFederationMetadataLocation which is not existing in the vNext version anymore.

**Comment by freddydk** [comment]
I guess this is only when you use AAD auth. right?

**Comment by freddydk** [comment]
You can add 
```
-myscripts @('https://raw.githubusercontent.com/microsoft/nav-docker/master/generic/Run/210-new/SetupConfiguration.ps1')
```
Until a new generic image is shipped

**Comment by nicolassaleron** [comment]
@kine Have you been able to fix the issue with the latest bcinsider? After adding the SetupConfiguration.ps1 script, the container is created but the client is not able to open a session.
I get this error in the browser:
```
You cannot sign in due to a technical issue. Contact your system administrator.
```

**Comment by freddydk** [comment]
That error is typically because you have a multitenant container and forgot to specify tenant?

**Comment by freddydk** [comment]
Maybe download the eventlog and see what the cause is?

**Comment by DanielGoehler** [comment]
@freddydk We have the same Problem with AAD Container. Eventlog doesn't say anything.

**Comment by kine** [comment]
> @kine Have you been able to fix the issue with the latest bcinsider? After adding the SetupConfiguration.ps1 script, the container is created but the client is not able to open a session. I get this error in the browser:
> 
> ```
> You cannot sign in due to a technical issue. Contact your system administrator.
> ```

If you are creating the container with AAD auth, you need to create the AAD app and set appropriate settings inside the container like:

```powershell
Set-NAVWebServerInstanceConfiguration -WebServerInstance BC -KeyName AadAuthorityUri -KeyValue 'https://login.microsoftonline.com/<tenantid>'
Set-NAVWebServerInstanceConfiguration -WebServerInstance BC -KeyName AadApplicationId -KeyValue '<appid>'
```

**Comment by kine** [comment]
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/administration/authenticating-users-with-azure-ad-overview

**Comment by freddydk** [comment]
I think they are available on New-BcContainer:

```
New-BcContainer @params `
    -containerName $containerName `
    -accept_eula `
    -artifact $artifactUrl `
    -auth AAD `
    -Credential $credential `
    -licenseFile $licenseFile `
    -updatehosts `
    -AuthenticationEMail $AadCredential.UserName `
    -AadTenant $aadTenant `
    -AadAppId $AdProperties.SsoAdAppId `
    -AadAppIdUri $appIdUri `
    -dns 'hostDNS' `
    -myScripts @('https://raw.githubusercontent.com/microsoft/nav-docker/master/generic/Run/210-new/SetupConfiguration.ps1') `
    -additionalParameters @("--env customNavSettings=ExcelAddInAzureActiveDirectoryClientId=$($AdProperties.ExcelAdAppId)")
```

**Comment by nicolassaleron** [comment]
I gonna log another issue, it does not seem directly linked with this one. Sorry for messing things :o

**Comment by freddydk** [comment]
The above script (from a few days ago) now works for BC 22 as well.
See #2876 for more information.

A new generic image version will be out in a few days with this fix (meaning you don't need the -myscripts anymore)

**Comment by freddydk** [comment]
Next containerHelper version will auto-patch SetupConfiguration if you are using AAD and generic 1.0.2.13 or earlier with 22.x and above
Real fix is with generic 1.0.2.14

**Comment by freddydk** [comment]
Shipped in 4.0.14 and Generic image 1.0.2.14
